### PR TITLE
[thorfinn] Round-4: Lion + width 768d (µP lr=3.3e-5) — architectural capacity uplift

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

Hyperparameter tuning on the 4L/512d/8h Transolver backbone has **plateaued** around the new Lion SOTA `vnb7oheo` (test_abupt **11.30**). 12 of 14 PRs in the last week have closed on regression; the only wins (PR #46 → 14.55, then SOTA `vnb7oheo` → 11.30) came from optimizer/coord-encoding changes, not from architectural changes within the existing 4L/512d capacity.

The binding gaps now are **volume_pressure ×2.1** and **tau_y/tau_z ×3.8/×3.9** — physics channels where the model's representational capacity is limited, not where its training is limited. **Doubling per-token capacity (512d → 768d) on the SOTA Lion recipe should let the model carry 6 distinct physics channels (p_s, ws_x/y/z, p_v) into the slice attention with 50% more headroom each.**

This is an **architectural single delta** on the active SOTA — width is the most-cited "first thing to try" lever (per-Transolver and AB-UPT papers both use 768d as the productive scale for scientific surrogates). yi advisor #18 explicitly directed: "be bolder; replace the backbone … mine noam/radford branches" — width 768d is the conservative architectural step.

## Why width and why 768d?

- **AB-UPT public reference uses 768d** for surface+volume regression on this exact dataset. We're at 512d — closing this gap to the reference architecture is overdue.
- 4L × 512d × 8h × 128 slices = **12.77M params**. 4L × 768d × 12h × 128 slices ≈ **28.8M params** (2.25× param count, 2.25× FLOPs). Lion's sign-update geometry means per-channel gradient magnitude is normalized — wider models do not destabilize Lion the way they destabilize AdamW.
- Width upgrade is orthogonal to RFF, vol_w, schedule fixes (all lower-yield levers being tested in #51, #54, #55, #57, #68). This is the missing axis.
- µP-scaling: scale `lr ← lr × (d_old/d_new)`. Going 512 → 768 means **lr ← 5e-5 × (512/768) = 3.33e-5**. This preserves the per-layer effective LR and prevents Lion's signs from over-stepping a wider feature space.

## Why 12 heads (not 8)?

To keep `head_dim = hidden/heads` constant: 512/8 = 64 = 768/12. This preserves the per-head attention spectrum (lower head_dim → sharper attention; we don't want to widen heads). 8h with 768d → 96 head_dim is anomalously sharp and rarely used.

## Single-delta from SOTA `vnb7oheo` reproduce

```bash
cd target/
torchrun --standalone --nproc_per_node=8 train.py \
  --agent thorfinn \
  --wandb-name "thorfinn/round4-lion-width768d" \
  --wandb-group "tay-round4-lion-width768d" \
  --optimizer lion --lion-beta1 0.9 --lion-beta2 0.99 \
  --lr 3.33e-5 --weight-decay 5e-4 \
  --volume-loss-weight 2.0 --batch-size 4 --validation-every 1 \
  --train-surface-points 65536 --eval-surface-points 65536 \
  --train-volume-points 65536 --eval-volume-points 65536 \
  --model-layers 4 --model-hidden-dim 768 --model-heads 12 --model-slices 128 \
  --ema-decay 0.9995 \
  --gradient-log-every 100 --weight-log-every 100 \
  --no-log-gradient-histograms
```

Effective single-delta from `vnb7oheo` (Lion lr=5e-5/wd=5e-4 SOTA):
- `--model-hidden-dim 512 → 768`
- `--model-heads 8 → 12` (preserves head_dim=64)
- `--lr 5e-5 → 3.33e-5` (µP scaling: lr ∝ 1/d)

**No code changes required.** Train.py already accepts these flags.

## Predictions

- **If the hypothesis is right (capacity-bound):** test_abupt should land in the **9.5–10.8** range (≈8-15% better than SOTA 11.30). The largest gain should be on volume_pressure (the channel with most physics complexity), followed by tau_y/tau_z. Surface_pressure (already ×1.6 from AB-UPT) has less headroom.
- **If the model is training-bound (not capacity-bound):** test should land flat (10.5-11.5) or slightly worse, because the wider model needs more data / more steps to find good generalization. In that case a follow-up at higher LR (or longer training) would close the gap.
- **Failure mode 1 — div / OOM:** if memory is tight on RTX PRO 6000 96GB, drop to `--train-volume-points 49152 --eval-volume-points 49152` (75% of current) — should reclaim ~7-8 GB.
- **Failure mode 2 — Lion divergence:** if val explodes after epoch 8-10 (the pattern from PRs #42 #52 #68), it means the µP-scaled LR is still too high for the wider model. Mid-flight kill, restart with lr=2.5e-5.

## What to include in your results comment

1. All `test_primary/*` metrics (abupt mean + all 6 axes) from best-val checkpoint reload.
2. Best-val abupt and best epoch number.
3. Total runtime (minutes) and effective epochs reached — confirm under 270 min.
4. **Per-axis comparison vs SOTA `vnb7oheo`** focused on volume_pressure / tau_y / tau_z — if the hypothesis is right, these should improve more than surface_pressure.
5. Param count of model (sanity check that width actually changed).
6. Peak GPU memory (we're closer to the 96GB ceiling now).
7. **Diagnostic if Lion diverges:** save the val_curve plot up to the divergence point + the train_loss / grad_norm at the divergence step.

## Why you for this experiment

Your PR #56 diagnostic was the cleanest analytical writeup we've had this round — you correctly identified that the "miscalibration" was load-bearing and that the model was undertrained, not overoptimized. The width experiment is now the highest-yield single-delta on the SOTA, and reasoning about whether the 768d gains come from capacity vs. changed effective LR will be the same kind of analytical work you just demonstrated.

## Baseline

| Metric | SOTA `vnb7oheo` | AB-UPT ref |
|---|---:|---:|
| `test/abupt_axis_mean_rel_l2_pct` | **11.303** | — |
| `test/surface_pressure_rel_l2_pct` | 6.216 | 3.82 |
| `test/wall_shear_rel_l2_pct` | 11.315 | 7.29 |
| `test/volume_pressure_rel_l2_pct` | 12.755 | 6.08 |
| `test/wall_shear_x_rel_l2_pct` | 9.563 | 5.35 |
| `test/wall_shear_y_rel_l2_pct` | 13.831 | 3.65 |
| `test/wall_shear_z_rel_l2_pct` | 14.147 | 3.63 |

The `vnb7oheo` SOTA was uncompiled. **Stay uncompiled** for this experiment — Lion+compile has shown a late-training divergence pattern across PRs #42, #52, #68, and likely #57 (best val 42.58 then 54.70 then 93.12 — diverged epoch 4). Width is the lever; compile is the saboteur. We'll layer compile back on after width is verified stable.

### CODE REVIEW

**Comment by morganmcg1** [comment]
STUDENT thorfinn:

OOM hit on initial attempts (07:42 UTC, 07:50 UTC) at the recipe's 65536+65536 surface/volume points — Tried to allocate 1.50 GiB on a GPU with 935 MiB free, ~94 GB already in use. This is exactly failure-mode-1 from your PR body.

Applying the prescribed fix: drop `--train-volume-points 49152 --eval-volume-points 49152` (surface unchanged at 65536). Per your estimate this should reclaim ~7-8 GB and stay below the 96 GB ceiling.

Will note volume-point change in the results comment so apples-to-apples remain visible. Also adding `--no-compile-model` explicitly per the "Stay uncompiled" note (the inline command snippet was missing it; default is True).

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #69 because ADVISOR: Closing — test_abupt = 12.351 (+9.3% vs new SOTA 11.208). 768d uncompiled doesn't deliver within the 270-min budget.

Results:
- W&B run: mmbry5md, rt=293m, 5 val epochs
- Val trajectory: 71.04→30.65→16.42→12.50→11.23 (strong descent but only 5 epochs)
- test_abupt = 12.351 vs SOTA 11.208 = +9.9% regression

Why 768d fails within budget:
- 768d is ~2.25× params vs 512d — each epoch takes ~65min vs ~30min uncompiled
- Only 5 epochs completed in 293min vs 9 epochs for 512d
- 5 epochs of 768d vs 9 epochs of 512d = less optimization trajectory

The val 11.23 at ep5 is comparable to 512d at ep7 (11.10), suggesting 768d may eventually converge lower — but it cannot within the 270-min budget. Architectural capacity lever needs either (a) compile to get more epochs, or (b) budget extension.

Key finding: **768d uncompiled is budget-limited, not capacity-limited.** If frieren #73 (6L compile) or a future 768d+compile run can fit more epochs, there's still architectural upside. Closing this specific config..
